packages/datadog-plugin-azure-functions/test/integration-test/client.spec.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const {
+  FakeAgent,
+  hookFile,
+  createSandbox,
+  curlAndAssertMessage
+} = require('../../../../integration-tests/helpers')
+const { spawn } = require('child_process')
+const { assert } = require('chai')
+const findProcess = require('find-process')
+const waitOn = require('wait-on')
+
+describe('esm', () => {
+  let agent
+  let proc
+  let sandbox
+
+  withVersions('azure-functions', '@azure/functions', version => {
+    before(async function () {
+      this.timeout(50000)
+      sandbox = await createSandbox([`@azure/functions@${version}`, 'azure-functions-core-tools@4'], false,
+        ['./packages/datadog-plugin-azure-functions/test/integration-test/fixtures/*'])
+    })
+
+    after(async function () {
+      this.timeout(50000)
+      await sandbox.remove()
+    })
+
+    beforeEach(async () => {
+      agent = await new FakeAgent().start()
+    })
+
+    afterEach(async () => {
+      const azureFuncProc = await findProcess('name', 'func', true)
+      const azureFuncProcPid = azureFuncProc[0]?.pid ?? null
+      azureFuncProcPid !== null && process.kill(azureFuncProcPid, 'SIGKILL')
+
+      proc && proc.kill()
+      await agent.stop()
+    })
+
+    it('is instrumented', async () => {
+      const envArgs = {
+        PATH: `${sandbox.folder}/node_modules/.bin:${process.env.PATH}`
+      }
+      proc = await spawnPluginIntegrationTestProc(sandbox.folder, 'func', ['start'], agent.port, undefined, envArgs)
+
+      return curlAndAssertMessage(agent, 'http://127.0.0.1:7071/api/httptest', ({ headers, payload }) => {
+        assert.propertyVal(headers, 'host', `127.0.0.1:${agent.port}`)
+        assert.isArray(payload)
+        assert.strictEqual(payload.length, 1)
+        assert.isArray(payload[0])
+        assert.strictEqual(payload[0].length, 1)
+        assert.propertyVal(payload[0][0], 'name', 'azure-functions.invoke')
+      })
+    }).timeout(50000)
+  })
+})
+
+async function spawnPluginIntegrationTestProc (cwd, command, args, agentPort, stdioHandler, additionalEnvArgs = {}) {
+  let env = {
+    NODE_OPTIONS: `--loader=${hookFile}`,
+    DD_TRACE_AGENT_PORT: agentPort
+  }
+  env = { ...env, ...additionalEnvArgs }
+  return spawnProc(command, args, {
+    cwd,
+    env
+  }, stdioHandler)
+}
+
+function spawnProc (command, args, options = {}, stdioHandler, stderrHandler) {
+  const proc = spawn(command, args, { ...options, stdio: 'pipe' })
+  return new Promise((resolve, reject) => {
+    waitOn({
```

Comment:
Why is this needed? It adds a dependency and doesn't really seem necessary since the rest of the code just waits for output.

Reply:
Are you asking if `waitOn` is necessary? Without waiting for the resource to be up there is this connection error.

```
Error: connect ECONNREFUSED 127.0.0.1:7071
```

Reply:
Why though? Can't the child process just signal to the parent process when it's ready instead of having to poll using a new dependency?

Reply:
I removed the `wait-on` dependency by checking for when the output from the `func start` command shows that the endpoint is ready. Let me know if that's what you meant.

https://github.com/DataDog/dd-trace-js/blob/8a81f3ce5e4fd6c6f6f7a99084284d03b77f7c71/packages/datadog-plugin-azure-functions/test/integration-test/client.spec.js#L92-L94